modules/mqnic/mqnic_selq.c

```diff
@@ -0,0 +1,223 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Views */
+/*
+ * Copyright 2021, Missing Link Electronics, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above
+ *       copyright notice, this list of conditions and the following
+ *       disclaimer in the documentation and/or other materials provided
+ *       with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation
+ * are those of the authors and should not be interpreted as representing
+ * official policies, either expressed or implied, of Missing Link
+ * Electronics, Inc.
+ */
+
+#include <linux/module.h>
+#include <linux/netdevice.h>
+#include <linux/rtnetlink.h>
+#include <linux/skbuff.h>
+#include <linux/version.h>
+
+#include "mqnic.h"
+#include "mqnic_selq.h"
+
+bool mqnic_selq_handler_enable = true;
+
+module_param_named(selq_handler_enable, mqnic_selq_handler_enable, bool, 0444);
+MODULE_PARM_DESC(selq_handler_enable, "enable select queue handler capability (default: true)");
+
+#define mqnic_is_selq_handler_available(ndev) \
+	(ndev->ethtool_ops == &mqnic_ethtool_ops)
```

Comment:
This doesn't look right...should this be `ndev->netdev_ops == &mqnic_selq_netdev_ops` instead?

Reply:
Since the purpose of this is to identify if a particular net_device actually is an mqnic net_device, any net_device member which is specific/unqiue to mqnic is sufficient. Of course, I was kind of lazy in picking that symbol `mqnic_ethtool_ops` for identification - simple reason: it was already there is `mqnic.h` ;-). So as long as symbol `mqnic_ethtool_ops` is specific to mqnic, the purpose of identification is full-filled the same way as it would be with any other specific symbol/`struct net_device` member.

However I also think now that using the `netdev_ops` makes more sense, cleaner, since we are dealing with a `netdev_ops` member for selq anyway. So I'll update this.